### ISSUE TITLE
Feature request: Intersection.

### ISSUE THREAD

**Issue by SreeKrishnaRaja** [coderz-mafia/caninejs#23]
Should compute the intersection of input arrays.

If the input is array of objects or array of maps, it should compute intersection based on an input key.

Examples:
If inputs are simple arrays.
Syntax: intersection(array, array, ...);

```
intersection([0, 1, 2], [2, 4, 1, 8]);
// [1, 2]
```
If inputs are array of objects or maps.
Syntax: intersection(array, array, ... , key)

```
union([{ id: 1, name: 'krishna' }, { id: 2, name: 'jk' }], [{ id: 1, name: 'krishna' }, { id: 3, name: 'justin' }], 'id')
// [{ id: 1, name: 'krishna' }]
```

**Comment by SreeKrishnaRaja** [comment]
Feature is implemented in #27

**Comment by SreeKrishnaRaja** [comment]
#27 is merged with master
